src/raw_rwlock.rs

```diff
@@ -346,8 +346,8 @@ unsafe impl lock_api::RawRwLockUpgrade for RawRwLock {
     unsafe fn unlock_upgradable(&self) {
         self.deadlock_release();
         let state = self.state.load(Ordering::Relaxed);
-        if state & PARKED_BIT == 0 {
```

Comment:
Please don't do this, it makes the code less readable.

Reply:
It was a clippy warning to keep it as-is.
IMO it's not less readable.

Reply:
I understand that this comes from a clippy lint, but I not particularly fond of that lint. The `if` statements are intentionally nested this way so that the structure of the code is clearer. If you must, add an `#[allow]` to disable this clippy lint.

Reply:
Updated to match your preferences. 👍